### PR TITLE
Start dogfooding editorconfig support in the branch

### PR DESCRIPTION
**Pull request by agocke**
Switches our LKG compiler to a build from the editorconfig branch and moves a diagnostic to be suppressed through editorconfig to test functionality.

### CODE REVIEW

File: eng/config/rulesets/Shipping.ruleset [hunk 1]
```diff
@@ -154,8 +154,6 @@
     <Rule Id="xUnit2013" Action="None" /> <!-- "do not use Assert.Equal() to check for collection size" is a valid assert, but very noisy right now -->
     <Rule Id="xUnit2015" Action="None" /> <!-- "do not use typeof() expression to check the exception type. " is a valid assert, but very noisy right now -->
     <Rule Id="xUnit2017" Action="None" /> <!-- "do not use Contains() to check if a value exists in a collection" is a valid assert, but very noisy right now -->
-    <Rule Id="xUnit2018" Action="None" /> <!-- "do not compare an object's exact type to the abstract class" is a valid assert, but very noisy right now -->
-collection.
```
Comment (jaredpar) [inline eng/config/rulesets/Shipping.ruleset]: Why is this being removed?

Comment (jaredpar) [reply eng/config/rulesets/Shipping.ruleset]: Nevermind

Comment (agocke) [reply eng/config/rulesets/Shipping.ruleset]: Just moved. I moved it into the editorconfig file so I could suppress it using editorconfig and verify things were working.

**Comment by Pilchie** [comment]
This is awesome!

**Pull request merged**